### ISSUE TITLE
metadata annotations too long in cephnfses CR

### ISSUE THREAD

**Issue by yuvalif** [rook/rook#11949]
when deploying rook operator (from master). running:
```
kubectl apply -f crds.yaml
```
gives the following error message:
```
The CustomResourceDefinition "cephnfses.ceph.rook.io" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

**Environment**:
* OS (e.g. from /etc/os-release): Red Hat Enterprise Linux 8.6 (Ootpa)
* Kernel (e.g. `uname -a`): 4.18.0-372.32.1.el8_6.x86_64
* Rook version (use `rook version` inside of a Rook Pod): master branch
* Kubernetes version (use `kubectl version`): v1.26.1

**Comment by yuvalif** [comment]
not sure why the complaint, annotation looks small
```yaml
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  annotations:
    controller-gen.kubebuilder.io/version: v0.11.3
  creationTimestamp: null
  name: cephnfses.ceph.rook.io          
```

**Comment by travisn** [comment]
Try `kubectl replace -f crds.yaml`, see #11772 for a recent doc change. Definitely annoying that we're hitting this limit though.

**Comment by travisn** [comment]
> not sure why the complaint, annotation looks small

It maybe complaining about the size of the crds.yaml file

**Comment by yuvalif** [comment]
> Try `kubectl replace -f crds.yaml`, see #11772 for a recent doc change. Definitely annoying that we're hitting this limit though.

i want to install, not upgrade. cannot replace something which is not there

**Comment by brightdroid** [comment]
Same problem here on k3s while updating rook-ceph to 1.11.2 as helm chart with argocd:

> CustomResourceDefinition.apiextensions.k8s.io "cephnfses.ceph.rook.io" is invalid: metadata.annotations: Too long: must have at most 262144 bytes

**Comment by brightdroid** [comment]
Could fix the problem by using Argo CD syncOption "ServerSideApply=true"

**Comment by travisn** [comment]
This was fixed by #12155
